src/core_editor/line_buffer.rs

```diff
@@ -206,11 +200,11 @@ impl LineBuffer {
     where
```

Comment:
Given that this function operates on the current line maybe its name should reflect that. Thoughts?
`replace_range_in_current_line`.
Well ideally this comment applies to all the functions that operate on current line.

Reply:
I moved the functionality away from thinking in lines to thinking in ranges in the buffer. Offsets then are from the start of the buffer rather than the start of the line.

I wasn't able to find anywhere where the logic in the line editing part of reedline assumed a kind of 2-dimentional edit, but seemingly almost always just worked on the buffer given.